hyperspy/io_plugins/bcf.py

```diff
@@ -677,22 +677,26 @@ def _parse_image(self, xml_node, overview=False):
         image = Container()
         image.width = int(xml_node.find('./Width').text)  # in pixels
         image.height = int(xml_node.find('./Height').text)  # in pixels
+        image.dtype = 'u' + xml_node.find('./ItemSize').text 
         image.plane_count = int(xml_node.find('./PlaneCount').text)
         image.images = []
         for i in range(image.plane_count):
             img = xml_node.find("./Plane" + str(i))
             raw = codecs.decode((img.find('./Data').text).encode('ascii'),'base64')
-            array1 = np.frombuffer(raw, dtype=np.uint16)
+            array1 = np.frombuffer(raw, dtype=image.dtype)
             if any(array1):
                 item = self.gen_hspy_item_dict_basic()
                 data = array1.reshape((image.height, image.width))
-                detector_name = str(img.find('./Description').text)
+                desc = img.find('./Description')
+                #detector_name = str(dscrp.text) if  else ''
                 item['data'] = data
                 item['axes'][0]['size'] = image.height
                 item['axes'][1]['size'] = image.width
-                item['metadata']['General'] = {'title': detector_name}
-                item['metadata']['Signal'] = {'signal_type': detector_name,
-                                              'record_by': 'image'}
+                item['metadata']['Signal'] = {'record_by': 'image'}
+                item['metadata']['General'] = {}
+                if desc is not None:
+                    item['metadata']['Signal']['signal_type'] = str(desc.text)
```

Comment:
The `signal_type` is not supposed to be set with this. I think if you don't have the correct metadata to put there, then just ignore it. The information is already in the title just below, so this is enough.

Reply:
Esprit (Bruker's software) behaves differently on different microscopes. It have two channels for getting SEM images (getting images by ext scan using the SEM detectors). Some instruments have intelligent link of Esprit, and it is able to get the signal name correctly; other instruments have limited link with SEM, and have no way in knowing what type of detectors is set for channel 1 and channel 2. For such a case, Esprit gui allows to provide it manually under `image` setup tab. If user is not aware of that, the signals are by default named as `Ch 1` and `Ch 2` (or `signal 1` and `signal 2`, or other linguistic variants, which is). In many cases User sets those names accordingly (e.g. SEI, BSE). In some cases it can get out of sync (it is up to user to keep it named rightly, if system is not intelligent). However there can be more than 2 images saved in bcf, and normally signals of 3rd and more images are named automatically (e.g. `Argus` detector when EBSD detector is inserted). All of these use tag <Description> and looking to other libraries and examples I thought that that should be mapped to `metadata.Signal.signal_type`. But I relooked the Metadata structure in documentation and you are right. SEI, BSE, Argus... should not go there, only EDS_TEM and EDS_SEM; I don't know how I missed this.

Generally I see that `Metadata structure`  in docs got quite much updated (there is such thing as Stage) I think it will be good idea to update bcf spec and map some of available metadata into hyperspy signals metadata.